R/teal_lockfile.R

```diff
@@ -39,50 +39,54 @@ teal_lockfile <- function() {
     }
   }
 
-  if (!(is_in_test() || is_r_cmd_check())) {
-    old_plan <- future::plan()
-    # If there is already a parallel (non-sequential) backend, reuse it.
-    if (inherits(old_plan, "sequential")) {
-      future::plan(future::multisession, workers = 2)
-    }
+  shiny::onStop(function() file.remove(lockfile_path))
 
-    lockfile_task <- ExtendedTask$new(create_renv_lockfile)
-    lockfile_task$invoke(close = inherits(old_plan, "sequential"), lockfile_path)
-    logger::log_trace("lockfile creation invoked.")
-  }
+  # Both capture.output are not needed if stdout = "|"
+  callr_output <-
+    utils::capture.output( # Needed to suppress: 'Opening fd 1' message
+      utils::capture.output( # Needed to suppress: 'PROCESS 'Rterm', running, pid' output
+        callr::r_bg(
+          func = create_renv_lockfile,
+          args = list(
+            lockfile_path = lockfile_path,
+            opts = options()
+          ),
+          # print results to the console
+          stdout = "",
```

Comment:
I don't really follow here.
We print to the console and capture the output not to print to console but to save to var instead but that var is nowhere used.

```suggestion
  callr_output <-
    utils::capture.output( # Needed to suppress: 'Opening fd 1' message
      utils::capture.output( # Needed to suppress: 'PROCESS 'Rterm', running, pid' output
        callr::r_bg(
          func = create_renv_lockfile,
          args = list(
            lockfile_path = lockfile_path,
            opts = options()
          ),
          # print results to the console
          stdout = "",
```

Reply:
What is returned by `callr::r_bg(` with `stdout = ""` is printed directly to the console through `callr` process. On top of it, main session print a message and a regular text to the console. With two `capture.outputs` we disable what main session prints to the console. This way we only get what `callr` process printed. However I don't think `callr_output <-` is necessary.

This whole logging thing might not be working while shiny will be occupying the console output during the runtime. I am investigating this in https://github.com/insightsengineering/teal/pull/1255#discussion_r1662467473

Reply:
Removed the var and one `caputre.output` as it was not needed in the end e7c7c7b5fbea4f8850e07f3d1c445e4a77a348d5